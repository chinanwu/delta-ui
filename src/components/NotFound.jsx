import React from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';

import getThemeClassname from '../functions/getThemeClassname';

import withTitle from './HOC/withTitle.jsx';
import ThemeToggle from './ThemeToggle.jsx';

import './NotFound.less';

export const NotFound = ({ dark }) => (
	<div className={getThemeClassname('NotFound', dark)}>
		<ThemeToggle />
		<h1>404 - Page Not Found!</h1>
		<h2 className="NotFound__linkContainer">
			<span className="NotFound__linkWords">Perhaps you'd like a link</span>
			<Link className={getThemeClassname('NotFound__link', dark)} to="/">
				home
			</Link>
			?
		</h2>
	</div>
);

export const mapStateToProps = ({ theme: { dark } }) => ({
	dark,
});

export default withTitle('Not Found')(connect(mapStateToProps)(NotFound));
