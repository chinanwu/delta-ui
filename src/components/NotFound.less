.NotFound {
	height: 100%;
	width: 100%;
	display: flex;
	flex-direction: column;
	justify-content: center;
	align-items: center;
}

.NotFound__link {
	display: flex;
}

.NotFound__linkWords::after {
	content: '\00a0';
}
