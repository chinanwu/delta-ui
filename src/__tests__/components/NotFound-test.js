import { shallow } from 'enzyme';
import React from 'react';

import { NotFound, mapStateToProps } from '../../components/NotFound.jsx';

jest.unmock('../../components/NotFound.jsx');
jest.unmock('../../functions/getThemeClassname');

// mock the HOC
jest.mock('../../components/HOC/withTitle', () => () => Component => props => (
	<Component {...props} />
));

describe('NotFound component', () => {
	describe('rendering', () => {
		it('renders NotFound', () => {
			const wrapper = shallow(<NotFound />);
			expect(wrapper.getElement()).toMatchSnapshot();
		});
	});

	describe('mapStateToProps', () => {
		it('returned mapped properties', () => {
			const theme = { dark: true };
			const state = { theme };
			expect(mapStateToProps(state)).toEqual(theme);
		});
	});
});
