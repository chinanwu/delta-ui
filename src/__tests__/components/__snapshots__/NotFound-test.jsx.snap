// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFound component rendering renders NavBar 1`] = `
<div>
  404!
</div>
`;

exports[`NotFound component rendering renders NotFound 1`] = `
<div>
  <Memo(Connect(ThemeToggle)) />
  <h1>
    404, Page Not Found!
  </h1>
  <h2
    className="NotFound__link"
  >
    <span
      className="NotFound__linkWords"
    >
      Perhaps you'd like a link
    </span>
    <ForwardRef
      to="/"
    >
      home
    </ForwardRef>
    ?
  </h2>
</div>
`;
